march_rqt_gait_generator/src/march_rqt_gait_generator/joint_plot.py

```diff
@@ -99,9 +101,23 @@ def create_plots(self, joint):
         self.plot_min_effort = self.plot(pen=pg.mkPen(color='r'))
         self.plot_max_effort = self.plot(pen=pg.mkPen(color='r'))
 
-    def update_set_points(self, joint, show_velocity_plot=False, show_effort_plot=False):
+    def to_setpoints(self):
+        plot_data = self.plot_item.getData()
+        setpoints = []
+        for i in range(0, len(plot_data[0])):
+            velocity = self.velocities[i]
+            time = plot_data[0][i]
+            position = math.radians(plot_data[1][i])
+            setpoints.append(ModifiableSetpoint(time, position, velocity))
+        return setpoints
+
+    def update_setpoints(self, joint, show_velocity_plot=False, show_effort_plot=False):
         time, position, velocity = joint.get_setpoints_unzipped()
 
+        self.duration = joint.duration
+        self.setXRange(-0.1, self.duration + 0.1, padding=0)
```

Comment:
Where do `-0.1` and `0.1` come from?

Reply:
The setpoints at the border are not plotted if you let it go from exactly border to border. Hence, we set a small margin on both sides. Arguably, however, this should be a function of duration instead, like 1% of the duration. I will change it to that.

Reply:
Yes thank you for the explanation